src/api/spec/controllers/webui/cloud/azure/upload_jobs_controller_spec.rb

```diff
@@ -0,0 +1,230 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe Webui::Cloud::Azure::UploadJobsController, type: :controller, vcr: true do
+  let(:azure_configuration) { build(:azure_configuration, application_id: 'Hey OBS!', application_key: 'Hey OBS?') }
+  let(:user_with_azure_configuration) { create(:confirmed_user, login: 'tom', azure_configuration: azure_configuration) }
+  let(:project) { create(:project, name: 'Apache') }
+  let!(:package) { create(:package, name: 'apache2', project: project) }
+  let(:upload_job) { create(:upload_job, user: user_with_azure_configuration) }
+  let(:xml_response) do
+    <<-HEREDOC
+    <clouduploadjob name="6">
+      <state>created</state>
+      <details>waiting to receive image</details>
+      <created>1513604055</created>
+      <user>mlschroe</user>
+      <target>azure</target>
+      <project>Base:System</project>
+      <repository>openSUSE_Factory</repository>
+      <package>rpm</package>
+      <arch>x86_64</arch>
+      <filename>rpm-4.14.0-504.2.x86_64.rpm</filename>
+      <size>1690860</size>
+    </clouduploadjob>
+    HEREDOC
+  end
+  let(:public_key) { File.read(Rails.root.join('spec', 'support', 'files', 'cloudupload_public_key.txt')) }
+
+  before do
+    stub_request(:get, "#{CONFIG['source_url']}/cloudupload/_pubkey").and_return(body: public_key)
+    login(user_with_azure_configuration)
+  end
+
+  describe 'GET #new' do
+    context 'with valid parameters' do
+      before do
+        Feature.run_with_activated(:cloud_upload, :cloud_upload_azure) do
+          get :new, params: { project: 'Apache', package: 'apache2', repository: 'standard', arch: 'x86_64', filename: 'appliance.raw.xz' }
+        end
+      end
+
+      it { expect(response).to be_success }
+      it {
+        expect(assigns(:upload_job)).
+          to have_attributes(project: 'Apache', package: 'apache2', repository: 'standard', arch: 'x86_64', filename: 'appliance.raw.xz')
```

Comment:
What are you testing and why? ruby variable assingment? OpenStruct.new?

Reply:
What should I do? remove both (I mean for Ec2 too)? @ChrisBr do you want to say something? :smile:

Reply:
It is not a struct, it is a backend file. I would test that it is not nil!